networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/libp2p/rpc/RpcHandler.java

```diff
@@ -56,17 +57,21 @@ public class RpcHandler<
         TRequest,
         TRespHandler extends RpcResponseHandler<?>>
     implements ProtocolBinding<Controller<TOutgoingHandler>> {
-  private static final Duration TIMEOUT = Duration.ofSeconds(5);
   private static final Logger LOG = LogManager.getLogger();
 
-  private final RpcMethod<TOutgoingHandler, TRequest, TRespHandler> rpcMethod;
+  private static final Duration STREAM_INITIALIZE_TIMEOUT = Duration.ofSeconds(5);
+
   private final AsyncRunner asyncRunner;
+  private final RpcMethod<TOutgoingHandler, TRequest, TRespHandler> rpcMethod;
+  private final ThrottlingTaskQueue concurrentRequestsQueue;
 
   public RpcHandler(
       final AsyncRunner asyncRunner,
-      final RpcMethod<TOutgoingHandler, TRequest, TRespHandler> rpcMethod) {
+      final RpcMethod<TOutgoingHandler, TRequest, TRespHandler> rpcMethod,
+      final int maxConcurrentRequests) {
     this.asyncRunner = asyncRunner;
     this.rpcMethod = rpcMethod;
+    concurrentRequestsQueue = ThrottlingTaskQueue.create(maxConcurrentRequests);
```

Comment:
As far as I understand here we are limiting ourselves for max concurrent outgoing requests? Do we have the code which enforces inbound limit?

Reply:
Do you mean `The responder MAY penalise peers that concurrently open more than `MAX_CONCURRENT_REQUESTS` streams for the same request type, for the protocol IDs defined in this specification.` ? I was thinking it could be an incremental change and do it in a further PR since it is MAY instead of SHOULD/MUST.

Reply:
Oh, right! Fair point!